src/ec2/model.ts

```diff
@@ -41,21 +42,39 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
-        if (!IamRole) {
+        if (!IamRole?.Arn) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedPolicies = await this.iamClient.listAttachedRolePolicies(IamRole.Arn)
+            return attachedPolicies
+        } catch (e) {
+            const errorMessage = `No policies attached to role: ${IamRole.Arn}.`
+            getLogger().error(`ec2: ${errorMessage}`)
+            throw ToolkitError.chain(e, errorMessage, { code: 'NoSuchEntity' })
+        }
     }
 
     public async hasProperPolicies(instanceId: string): Promise<boolean> {
-        const attachedPolicies = (await this.getAttachedPolicies(instanceId)).map(policy => policy.PolicyName!)
-        const requiredPolicies = ['AmazonSSMManagedInstanceCore', 'AmazonSSMManagedEC2InstanceDefaultPolicy']
-
-        return requiredPolicies.length !== 0 && requiredPolicies.every(policy => attachedPolicies.includes(policy))
+        try {
+            const attachedPolicies = (await this.getAttachedPolicies(instanceId)).map(policy => policy.PolicyName!)
+            const requiredPolicies = ['AmazonSSMManagedInstanceCore', 'AmazonSSMManagedEC2InstanceDefaultPolicy']
+
+            return requiredPolicies.length !== 0 && requiredPolicies.every(policy => attachedPolicies.includes(policy))
+        } catch (e) {
+            if (e instanceof ToolkitError && e.code == 'NoSuchEntity') {
+                getLogger().warn(
+                    `ec2: due to error in checking policies attached to instance, assuming necessary policies do not exist for instance ${instanceId}.`
+                )
+                return false
```

Comment:
It looks like we are still silencing the error. We _want_ to show the user an error like:

    Role "foo" attached to EC2 instance has no policies attached.

Are we doing that?

Reply:
Oh, yes I was silencing it to show the same general error of lacking permissions. But just changed it to throw the more specific error instead.